🐛 Template changes/fixes

dharmit opened the pull request:
This PR does two things:
* Allow user to choose namespace instead of deploying everything into `default` ns
* Use `address.Address` instead of `address` in the Go template

dharmit committed:
Replace address with backend.Address

This is based on the error message and what can be found in this
template:

    https://github.com/rancher/cluster-api-provider-rke2/blob/9f64dac33063f1b699ae8dbeac56912864dd2b35/examples/docker/online-default/cluster-template.yaml

Signed-off-by: Dharmit Shah <dharmit.shah@suse.com>

anmazzotti commented on docker-rke2.yaml:
```diff
@@ -156,7 +156,7 @@ data:
       # TODO: we should be verifying (!)
       {{range $server, $address := .BackendServers}}
-      server {{ $server }} {{ JoinHostPort $address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
+      server {{ $server }} {{ JoinHostPort $address.Address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
```
nice finding!

Danil-Grigorev replied:
```suggestion
      {{range $server, $backend := .BackendServers }}
      server {{ $server }} {{ JoinHostPort $backend.Address $.BackendControlPlanePort }} check check-ssl verify none resolvers docker resolve-prefer {{ if $.IPv6 -}} ipv6 {{- else -}} ipv4 {{- end }}
      {{- end}}
```
to align with the template in the turtles repo: https://github.com/rancher/turtles/blob/85e718b7ef4d78cf6273dab603dda0fac9527c24/test/e2e/data/cluster-templates/docker-rke2.yaml

Maybe also it needs to be mentioned that this change works with CAPI 1.9+ only

anmazzotti reviewed (changes requested):
May you also add namespace to `Cluster` and `DockerCluster` please? I think this is a nice occasion to fix it.

dharmit committed:
Allow setting namespace for the resources

Everything's using `default` namespace (hard-coded). Allow users to
choose a namespace.

Signed-off-by: Dharmit Shah <dharmit.shah@suse.com>

dharmit commented:
> May you also add namespace to `Cluster` and `DockerCluster` please? I think this is a nice occasion to fix it.

Done. I have also set the namespace to default for all resources. Let me know if I misinterpreted your comment here https://github.com/rancher/turtles-docs/pull/204#discussion_r1928533260.

anmazzotti commented:
> > May you also add namespace to `Cluster` and `DockerCluster` please? I think this is a nice occasion to fix it.
> 
> Done. I have also set the namespace to default for all resources. Let me know if I misinterpreted your comment here [rancher/turtles-docs#204 (comment)](https://github.com/rancher/turtles-docs/pull/204#discussion_r1928533260).

Yes, sorry I was not clear enough. 
The template can use `namespace: ${NAMESPACE}`, I think that's nice since it allows the user to change it.

What I meant in the other comment, is to then use `export NAMESPACE=default`, so that there is no need to alert the user about creating a new namespace and adding an extra step.

Danil-Grigorev commented on docker-rke2.yaml:
```diff
@@ -156,7 +158,7 @@ data:
       http-check expect status 401
```
Also we can probably add fixes for LB healthcheck to use `200` code here, which will check for `ETCD` health as well - https://github.com/rancher/cluster-api-provider-rke2/commit/bb2410b647c8189520df78efe0e8c7a72f6394cf

dharmit committed:
Use status code 200 for LB healtcheck

Signed-off-by: Dharmit Shah <dharmit.shah@suse.com>

dharmit committed:
Enable anonymous auth in RKE2 control plane

kubeadm has anonymous auth enabled by default, but RKE2 disables it if
not overridden.

Signed-off-by: Dharmit Shah <dharmit.shah@suse.com>

dharmit committed:
Empty spec for RKE2ConfigTemplate

Signed-off-by: Dharmit Shah <dharmit.shah@suse.com>
